modules/platforms/cpp/thin-client/include/ignite/thin/transactions/transactions.h

```diff
@@ -0,0 +1,191 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef TRANSACTIONS_H
+#define TRANSACTIONS_H
+
+#include <string>
+
+#include <ignite/common/concurrent.h>
+#include <ignite/impl/thin/transactions/transactions_proxy.h>
+
+using namespace ignite::impl::thin::transactions;
+using namespace ignite::common::concurrent;
+
+namespace ignite
+{
+    namespace thin
+    {
+        namespace transactions
+        {
+            /**
+             * Transaction client.
+             *
+             * Implements main transactionsl API.
+             *
+             * This class implemented as a reference to an implementation so copying of this class instance will only
+             * create another reference to the same underlying object. Underlying object released automatically once all
+             * the instances are destructed.
+             */
+            class ClientTransaction {
+
+            public:                
+                ClientTransaction(TransactionProxy impl) :
+                    proxy(impl)
+                {}
+
+                /**
+                 * Destructor.
+                 */
+                ~ClientTransaction() {}
+
+                /**
+                 * Commits this transaction.
+                 */
+                void Commit()
+                {
+                    proxy.commit();
+                }
+
+                /**
+                 * Rolls back this transaction.
+                 */
+                void Rollback()
+                {
+                    proxy.rollback();
+                }
+
+                /**
+                 * Ends the transaction. Transaction will be rolled back if it has not been committed.
+                 */
+                void Close()
+                {
+                    proxy.close();
+                }
+
+                /**
+                 * Assignment operator.
+                 *
+                 * @param other Another instance.
+                 * @return *this.
+                 */
+                ClientTransaction& operator=(const ClientTransaction& other)
```

Comment:
If there is assignment operator, there should also be copy constructor.

Reply:
i remove it, looks like it excessive here.